containers_test.go

```diff
@@ -30,6 +30,45 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+func TestExecExitError(t *testing.T) {
+	ctx := testlog.WithTB(context.Background(), t)
+	client := DockerClient()
+	container, err := client.CreateContainer(docker.CreateContainerOptions{
+		Config: &docker.Config{
+			Image: "yourbase/yb_ubuntu:18.04",
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cmdStat := "stat /somewhere/strange/that/do/not/exist"
+	err = ExecShell(ctx, client, container.ID, cmdStat, &ExecShellOptions{
+		Interactive: true,
+	})
+	if err != nil {
+		if code, ok := IsExitError(err); !ok {
+			t.Errorf("Exit code: want 1, got %d", code)
```

Comment:
If the error doesn't match, the full error message is probably more helpful here:

```suggestion
		if code, ok := IsExitError(err); !ok || code != 1 {
			t.Errorf("error = %v; want exit code 1", err)
```

Reply:
It's possibly that I've hit moby/moby/issues/34488, still checking

Reply:
But I'm running 19.03.8-ce runc/moby, so I'm not entirely sure